src/cascadia/TerminalSettingsModel/Profile.h

```diff
@@ -125,6 +129,7 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         friend class SettingsModelLocalTests::KeyBindingsTests;
         friend class TerminalAppUnitTests::DynamicProfileTests;
         friend class TerminalAppUnitTests::JsonTests;
+        friend struct implementation::AppearanceConfig;

```

Comment:
Okay I might be missing something - why does AppearanceConfig need to be a friend of Profile? Is it doing anything with the internals of Profile?

Reply:
Yes, AppearanceConfig inherits from Profile, so it needs access to the internals

Reply:
I don't mean to be a pain, but what specifically is `AppearanceConfig` trying to access that is in `Profile`? It looks like `AppearanceConfig` is pretty self contained.

Reply:
AppearanceConfig has a profile as its parent, so whenever we try to access something that the appearance config hasn't defined, it will go ask its parent profile and try to call `_get##name##Impl()` which is a private member, so it needs to be friended

Reply:
wait hold up what now? `Profile` is a parent class of `AppearanceConfig`? That seems... backwards. So every `AppearanceConfig` instance has basically an entirely empty `Profile` as well? Including, technically, a null `AppearanceConfig`? That seems like a gross layering violation. Like a Profile should have an appearance, and possibly an unfocused appearance. I suppose the unfocused appearance should be a child of the profile's own appearance object, but the AppearanceConfig itself shouldn't need to know anything about the internals of a Profile.

Reply:
An `AppearanceConfig` has a `Profile` as a parent only in the sense of the Inheritance model, i.e. when we ask the appearance for a setting, if it itself didn't define it, then it will ask its parent profile for that setting. Unless I'm misunderstanding how `IInheritable` works, that does not mean that an `AppearanceConfig` has an empty `Profile` in it.

Reply:
nah see I'm thinking more like

![image](https://user-images.githubusercontent.com/18356694/109966314-bc006780-7cb5-11eb-9d7a-b35aeaac5852.png)

cause the `AppearanceConfig` _in theory_ shouldn't need to know anything about profiles. The profile should be responsible for hooking up it's unfocused `AppearanceConfig` to the "focused" `AppearanceConfig` in the profile. The unfocused one shouldn't think that it's parent is a `Profile`, it should just think that its parent is another `AppearanceConfig`.

Right?